From 31c2d7910c68d51bcbf1dd0d71c36cc593e07785 Mon Sep 17 00:00:00 2001
From: Dodji Seketeli <dodji@redhat.com>
Date: Wed, 1 Jul 2009 21:16:07 +0200
Subject: Fix crash when saving to a file (#587553)

	* src/mlview-old-gvc.cc (OldGVC::on_view_name_changed):
	Do not cast IView* into Gtk::Widget*. Rather use
	IView::get_view_widget().

diff --git a/src/mlview-old-gvc.cc b/src/mlview-old-gvc.cc
index 0feb6e1..9e6ab0c 100644
--- src/mlview-old-gvc.cc
+++ src/mlview-old-gvc.cc
@@ -292,7 +292,7 @@ OldGVC::on_view_name_changed (IView* a_view)
 
 	new_view_name = a_view->get_view_name () ;
 
-	view_impl = (Gtk::Widget*) a_view ;
+	view_impl = a_view->get_view_widget ();
 	THROW_IF_FAIL (view_impl) ;
 
 	view_container = view_impl->get_parent () ;
-- 
cgit v0.10.1

